pandas/tests/frame/test_indexing.py

```diff
@@ -3042,6 +3044,20 @@ def test_transpose(self):
         expected.index = ['A', 'B']
         assert_frame_equal(result, expected)
 
+    def test_scalar_assignment(self):
+        # issue #19843
+        df = pd.DataFrame(index=(0, 1, 2))
+        df['now'] = pd.Timestamp('20130101', tz='UTC')
+        assert isinstance(df.dtypes[0], DatetimeTZDtype)
+
+    def test_datetime_index_assignment(self):
```

Comment:
this is a duplicate test.

Reply:
you could put it as an additional test in the class as above if you'd like (parametrize as well)